### DISCUSSION TITLE
Running Linux ELF programs using Blink

### DISCUSSION THREAD

**Discussion by jankais3r** [holzschu/a-shell#604] [Ideas]
Hi, I am opening this thread to discuss experiments with [Blink](https://github.com/jart/blink) compiled to [WASM](https://github.com/jart/blink/issues/8#issuecomment-1383284995) in order to run regular Linux executables in a-shell. I don't currently have time to look into this myself, but I thought I would mention it here in case somebody wants to give it a try and report back.

**Comment by holzschu** [comment]
If the goal is to run regular X86 executables, I would  avoid adding an extra level of emulation. Compiling Blink directly to iOS/Arm64 would be more efficient, I think.

**Comment by HeavySnowJakarta** [comment]
> If the goal is to run regular X86 executables, I would  avoid adding an extra level of emulation. Compiling Blink directly to iOS/Arm64 would be more efficient, I think. 

I don't think it's unbearable to let it run with WebAssembly. Let's try it as a toy. It won't be a big problem with light items. If it's truly useful we can research to add it natively then.

**Comment by holzschu** [comment]
**Update:** I have (finally) cross-compiled `blink` to Arm64. That's only the beginning (I still have to deal with chroot, variables re-initialization, cleanup...) but that's a lot closer than it's ever been.
